common/sgx/quote.c

```diff
@@ -529,6 +546,227 @@ oe_result_t oe_verify_quote_with_sgx_endorsements(
+                collateral_expiration_status,
+                quote_verification_result,
+                supplemental_data,
+                supplemental_data_size_out,
+                qve_isvsvn_threshold),
+            "Failed to check QvE report and identity",
+            oe_result_str(result));
+
+        result = OE_OK;
+    }
+
+#else
+
+    result = sgx_verify_quote(
+        &_ecdsa_uuid,
+        NULL,
+        0,
+        quote,
+        (uint32_t)quote_size,
+        expiration_check_date,
+        &collateral_expiration_status,
+        &quote_verification_result,
+        NULL,
+        0,
+        supplemental_data,
+        MAX_SUPPLEMENTAL_DATA_SIZE,
+        &supplemental_data_size_out,
+        *(uint32_t*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_VERSION]
+                         .data),
+        (void*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_TCB_INFO]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_TCB_INFO].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_TCB_ISSUER_CHAIN]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_TCB_ISSUER_CHAIN].size,
+        (void*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_PROC_CA]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_PROC_CA].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_ISSUER_CHAIN_PCK_CERT]
+                    .data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_ISSUER_CHAIN_PCK_CERT]
+            .size,
+        (void*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_INFO]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_INFO].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_ISSUER_CHAIN]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_ISSUER_CHAIN]
+            .size);
+
+    if (result != OE_PLATFORM_ERROR)
+    {
+        OE_TRACE_INFO(
+            "SGX DCAP QVL-based quote verification is used, res: %s\n", result);
+
+        if (result != OE_OK)
+        {
+            OE_RAISE_MSG(
+                result,
+                "SGX QVL-based quote verification failed with error 0x%x",
+                result);
+        }
+
+        // Untrusted QVL use GMT, convert it to local time
+        p_supplemental_data = (sgx_qvl_supplemental_t*)supplemental_data;
+
+        OE_CHECK(oe_gmt_to_localtime(
+            &p_supplemental_data->earliest_expiration_date,
+            &p_supplemental_data->earliest_expiration_date));
```

Comment:
Why is this converted to local time?  Seems like any security checks should use UTC.

Reply:
Agree with @dthaler . The definition of the field states it's UTC:

    `time_t earliest_expiration_date;///< Earliest expiration date of all the collateral (UTC)`

Reply:
In step1, I just revised the implementation to use existing logic to get valid time range, so the function has been removed.